### PR TITLE
status codes for discovery client

### LINKED ISSUE #3502: Discovery Service: check API responses for correct status codes

### PR DESCRIPTION
**Pull request by gerardsn**
closes #3502 

PR contains
- add clientRegistrationManager.getServiceAndSubject to check for service and subject
- Add 404 for unknown subjectID and serviceID
- Add 202 on DELETE as documented in OAPI spec
- Log ignored errors
- cleanup `api/server#client.New`

### CODE REVIEW

File: discovery/client.go [hunk 1]
```diff
@@ -52,6 +52,8 @@ type clientRegistrationManager interface {
 	validate() error
 	// removeRevoked removes all revoked presentations from the store
 	removeRevoked() error
+	// getServiceAndSubject returns the service and subject, or ErrServiceNotFound / didsubject.ErrSubjectNotFound if either does not exist
+	getServiceAndSubject(ctx context.Context, serviceID, subjectID string) (ServiceDefinition, []did.DID, error)
```
Comment (woutslakhorst) [inline discovery/client.go:56]: why add it to the interface? There's no mock and only use is at the same struct.

File: discovery/client.go [hunk 2]
```diff
@@ -116,7 +110,7 @@ func (r *defaultClientRegistrationManager) activate(ctx context.Context, service
 	subjectDIDs = subjectDIDs[:j]
 
 	if len(subjectDIDs) == 0 {
-		return fmt.Errorf("%w: %w for %s", ErrPresentationRegistrationFailed, didsubject.ErrSubjectNotFound, subjectID)
+		return fmt.Errorf("%w: %w for %s", ErrPresentationRegistrationFailed, ErrDIDMethodsNotSupported, subjectID)
```
Comment (woutslakhorst) [inline discovery/client.go:113]: Better to change the error to something like: "no active DIDs found for subject"

**Review by woutslakhorst — CHANGES_REQUESTED** [review]

File: discovery/client.go (hunk 1 above)
Comment (gerardsn) [reply discovery/client.go:56]: removed interface since it only had 1 implementation

**Review by woutslakhorst — APPROVED** [review]

File: discovery/client.go (hunk 2 above)
Comment (gerardsn) [reply discovery/client.go]: added a new error for this with message `subject has no (active) DIDs matching the service`

**Pull request merged**